picard/config.py

```diff
@@ -129,6 +131,51 @@ def value(self, name, option_type, default=None):
         return default
 
 
+class SettingConfigSection(ConfigSection):
+    """Custom subclass to automatically accommodate saving and retrieving values based on user profile settings.
+    """
+    def __init__(self, config, name):
+        super().__init__(config, name)
+        self.__qt_config = config
+        self.__name = name
+        self.__prefix = self.__name + '/'
+        self._memoization = defaultdict(Memovar)
+        Option.add_if_missing(name, 'user_profiles', [])
+
+    def _get_profiles(self):
+        profiles = self['user_profiles']
+        return profiles if profiles is not None else []
+
+    def __getitem__(self, name):
+        # Don't process settings that are not profile-specific
+        if name in UserProfileGroups.get_profile_settings_list():
+            profiles = self._get_profiles()
+            for profile in profiles:
+                if profile['active']:
+                    value = getattr(profile['settings'], name, None)
+                    if value is not None:
+                        return value
+        opt = Option.get(self.__name, name)
+        if opt is None:
+            return None
+        return self.value(name, opt, opt.default)
+
+    def __setitem__(self, name, value):
+        # Don't process settings that are not profile-specific
+        if name in UserProfileGroups.get_profile_settings_list():
+            profiles = self._get_profiles()
+            for idx in range(len(profiles)):  # pylint: disable=consider-using-enumerate
```

Comment:
I don't get why we need to iterate using indexes here.

Reply:
Because I thought there would be a problem modifying the item in the list being iterated.  I just tested and found out that was an incorrect assumption.  Modified to remove the indexes.